### PR TITLE
[SPARK-17409] [SQL] [FOLLOW-UP] Do Not Optimize Query in CTAS More Than Once

### PR DESCRIPTION
**Pull request by gatorsmile**
### What changes were proposed in this pull request?

This follow-up PR is for addressing the [comment](https://github.com/apache/spark/pull/15048). 

We added two test cases based on the suggestion from @yhuai . One is a new test case using the `saveAsTable` API to create a data source table. Another is for CTAS on Hive serde table.

Note: No need to backport this PR to 2.0. Will submit a new PR to backport the whole fix with new test cases to Spark 2.0
### How was this patch tested?

N/A

### CODE REVIEW

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/MetastoreRelationSuite.scala [hunk 1]
```diff
@@ -36,4 +38,16 @@ class MetastoreRelationSuite extends SparkFunSuite {
     // No exception should be thrown
     relation.toJSON
   }
+
+  test("SPARK-17409: Do Not Optimize Query in CTAS More Than Once") {
+    withTable("bar") {
+      withTempView("foo") {
+        sql("select 0 as id").createOrReplaceTempView("foo")
+        sql("CREATE TABLE bar AS SELECT * FROM foo group by id")
```
Comment (yhuai) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/MetastoreRelationSuite.scala]: Also put the comment of https://github.com/apache/spark/pull/15459/files#diff-5d2ebf4e9ca5a990136b276859769289R1626 at here? Also mention that this test is for hive format table?

**Comment by gatorsmile** [comment]
@yhuai Any further comment about it? Thanks!

**Comment by gatorsmile** [comment]
cc @cloud-fan @yhuai This has been reviewed in the backport PR https://github.com/apache/spark/pull/15502. Can we merge this to master now? Thanks!

**Comment by cloud-fan** [comment]
LGTM

**Comment by cloud-fan** [comment]
merging to master, thanks!

**Pull request closed without merging**